spec.bs

```diff
@@ -405,6 +416,12 @@ fetch(issueRequest);
 To <dfn>append private state token issue request headers</dfn> given a [=/request=] |request|, run the following steps:
 
  1. If |request|'s [=request/client=] is not a [=secure context=], return.
+ 1. If |type| is not specified, throw a "{{TypeError}}" {{DOMException}}.
+ 1. If |type| is not `"private-state-token"`, throw a "{{TypeError}}" {{DOMException}}.
+ 1. If |version| is not specified, throw a "{{TypeError}}" {{DOMException}}.
+ 1. If |version| is not `1`, throw a "{{TypeError}}" {{DOMException}}.
+ 1. If |operation| is not specified, throw a "{{TypeError}}" {{DOMException}}.
+ 1. If |operation| is not one of {{OperationType}}, throw a "{{TypeError}}" {{DOMException}}.
```

Comment:
This should also not be necessary, see https://webidl.spec.whatwg.org/#es-enumeration

Reply:
However, it's not clear why this should be the only check on operation, i.e. is it valid to include `operation: "token-request"` on a redemption request? Probably not.

I suspect that these types are actually used at the start (i.e. in https://wicg.github.io/trust-token-api/#http-network-or-cache-fetch) to decide which operation to perform. Currently I think the spec would tell implementers to just do both.

Reply:
I do not think we need to add token type and version to the request. We already know them since they can only take one value. However, we do need to add the operation. We process the request based on the value of the operation field. 

1. https://source.chromium.org/chromium/chromium/src/+/main:services/network/url_loader_factory.cc;drc=089c96fe3d83e67d9a4fcc87241bad6a954ff143;l=273
2. https://source.chromium.org/chromium/chromium/src/+/main:services/network/trust_tokens/trust_token_request_helper_factory.cc;drc=bec0ef46974cd9a6bc0f1f1ae34baa11913f77af;l=140

Reply:
You're saying that you want those to be validated purely through the WebIDL constraints when they're passed in as part of the privateToken argument? That seems okay for now, but yeah, you'll still have to decide what to do based on the operation.

Reply:
Yes, verifying them through WebIDL constraints are good enough. I will add how operations and issuers are used. I think it might be better to add them in a different PR. Title of this one is specific.